luminoth/vis.py

```diff
@@ -0,0 +1,169 @@
+            coordinates[3] + i,
+        ]
+
+        # Fill must be drawn only for the first rectangle, or the alphas will
+        # add up.
+        if i == 0:
+            draw.rectangle(coords, fill=fill, outline=outline)
+        else:
+            draw.rectangle(coords, outline=outline)
+
+
+def draw_label(draw, coords, color, label, prob):
+    """Draw a box with the label and probability."""
+    # Attempt to get a native TTF font. If not, use the default bitmap font.
+    global SYSTEM_FONT
+    if SYSTEM_FONT:
+        label_font = SYSTEM_FONT.font_variant(size=16)
+        prob_font = SYSTEM_FONT.font_variant(size=12)
+    else:
+        label_font = ImageFont.load_default()
+        prob_font = ImageFont.load_default()
+
+    label = str(label)  # `label` may not be a string.
+    prob = '({:.2f})'.format(prob)  # Turn `prob` into a string.
+
+    # We want the probability font to be smaller, so we'll write the label in
+    # two steps.
+    label_w, label_h = label_font.getsize(label)
+    prob_w, prob_h = prob_font.getsize(prob)
+
+    # Get margins to manually adjust the spacing. The margin goes between each
+    # segment (i.e. margin, label, margin, prob, margin).
+    margin_w, margin_h = label_font.getsize('M')
+    margin_w *= 0.2
+    _, full_line_height = label_font.getsize('Mq')
+
+    # Draw the background first, considering all margins and the full line
+    # height.
+    background_coords = [
+        coords[0],
+        coords[1],
+        coords[0] + label_w + prob_w + 3 * margin_w,
+        coords[1] + full_line_height * 1.15,
+    ]
+    draw.rectangle(background_coords, fill=color + (255,))
+
+    # Then write the two pieces of text.
+    draw.text([
+        coords[0] + margin_w,
+        coords[1],
+    ], label, font=label_font)
+
+    draw.text([
+        coords[0] + label_w + 2 * margin_w,
+        coords[1] + (margin_h - prob_h),
+    ], prob, font=prob_font)
+
+
+def vis_objects(image, objects, colormap=None, labels=True):
+    """Visualize objects as returned by `Detector`.
+
+    Arguments:
+        image (numpy.ndarray): Image to draw the bounding boxes on.
+        objects (list of dicts or dict): List of objects as returned by a
+            `Detector` instance.
+        colormap (function): Colormap function to use for the objects.
+        labels (boolean): Whether to draw labels.
+
+    Returns:
+        A PIL image with the detected objects' bounding boxes and labels drawn.
+        Can be casted to a `numpy.ndarray` by using `numpy.array` on the
+        returned object.
+    """
+    if not isinstance(objects, list):
+        objects = [objects]
+
+    if colormap is None:
+        colormap = build_colormap()
+
+    image = Image.fromarray(image.astype(np.uint8))
+
+    draw = ImageDraw.Draw(image, 'RGBA')
+    for obj in objects:
+        # TODO: Can we do image resolution-agnostic?
+        color = colormap(obj['label'])
+        draw_rectangle(draw, obj['bbox'], color, width=3)
```

Comment:
It would be nice to have `width` (and `fill`) as options.

Reply:
Will add generic `scale` parameter which modifies the font size too, or sizes will be inconsistent.